src/main/java/com/artipie/settings/RepoData.java

```diff
@@ -0,0 +1,138 @@
+
+/**
+ * Repository data management.
+ * @since 0.1
+ */
+public final class RepoData {
+    /**
+     * Key 'storage' inside json-object.
+     */
+    private static final String STORAGE = "storage";
+
+    /**
+     * Repository settings storage.
+     */
+    private final Storage storage;
+
+    /**
+     * Ctor.
+     * @param storage Repository settings storage.
+     */
+    public RepoData(final Storage storage) {
+        this.storage = storage;
+    }
+
+    /**
+     * Remove data from the repository.
+     * @param rname Repository name
+     * @return Completable action of the remove operation
+     */
+    public CompletionStage<Void> remove(final RepositoryName rname) {
+        final String repo = rname.toString();
+        return this.asto(rname).deleteAll(new Key.From(repo)).thenAccept(
+            nothing -> Logger.info(this, String.format("Removed data from repository %s", repo))
+        );
+    }
+
+    /**
+     * Move data when repository is renamed: from location by the old name to location with
+     * new name.
+     * @param rname Repository name
+     * @param nname New repository name
+     * @return Completable action of the remove operation
+     */
+    public CompletionStage<Void> move(final RepositoryName rname, final RepositoryName nname) {
+        final Key repo = new Key.From(rname.toString());
+        final Key nrepo = new Key.From(nname.toString());
+        final Storage asto = this.asto(rname);
+        return new SubStorage(repo, asto).list(Key.ROOT).thenCompose(
+            list -> new Copy(new SubStorage(repo, asto), list)
+                .copy(new SubStorage(nrepo, asto))
+        ).thenCompose(nothing -> asto.deleteAll(new Key.From(repo))).thenAccept(
+            nothing ->
+                Logger.info(this, String.format("Moved data from repository %s to %s", repo, nrepo))
+        );
+    }
+
+    /**
+     * Obtain storage from repository settings.
+     * @param rname Repository name
+     * @return Abstract storage
+     * @throws UncheckedIOException On IO errors
+     */
+    private Storage asto(final RepositoryName rname) {
+        return new ConfigFile(String.format("%s.yaml", rname.toString()))
+            .valueFrom(this.storage)
+            .thenApply(PublisherAs::new)
+            .thenCompose(PublisherAs::asciiString)
+            .thenApply(Yaml::createYamlInput)
+            .thenApply(new UncheckedIOFunc<>(YamlInput::readYamlMapping))
+            .thenApply(yaml -> yaml.yamlMapping("repo"))
+            .thenApply(
+                yaml -> {
+                    YamlMapping res = yaml.yamlMapping(RepoData.STORAGE);
+                    if (res == null) {
+                        res = this.storageYamlByAlias(rname, yaml.string(RepoData.STORAGE));
+                    }
+                    return res;
+                }
+            )
+            .thenApply(yaml -> new YamlStorage(yaml).storage())
+            .toCompletableFuture().join();
```

Comment:
Blocking methods are not allowed to be called in the main code as they blocks threads. 
Also, I do not see `com.artipie.settings.StorageAliases#find` to be called and `StorageYamlConfig#storage` to be used. These methods already do all the job to create storage instance from yaml settings in respect with aliases. `StorageYamlConfig#storage` method is not needed, here is how this can be done:
```
/**
     * Obtain storage from repository settings.
     * @param rname Repository name
     * @return Abstract storage
     */
    private CompletionStage<Storage> repoStorage(final RepositoryName rname) {
        return new ConfigFile(String.format("%s.yaml", rname.toString()))
            .valueFrom(this.storage)
            .thenApply(PublisherAs::new)
            .thenCompose(PublisherAs::asciiString)
            .thenApply(Yaml::createYamlInput)
            .thenApply(new UncheckedIOFunc<>(YamlInput::readYamlMapping))
            .thenApply(yaml -> yaml.yamlMapping("repo").value("storage"))
            .thenCompose(
                yaml -> StorageAliases.find(this.storage, new Key.From(rname.toString())).thenApply(
                    aliases -> new StorageYamlConfig(yaml, aliases).storage()
                )
            );
    }
```
Note, that I return `CompletionStage` here, not `Storage` itself. You'll need to change `move` and `remove` a methods bit (just use `thenCompose` after calling `repoStorage` method, no major changes). 
Also not, that `yaml.yamlMapping("repo").value("storage")` here i call `value()` method to get storage yaml node as it can be scalar (if there is an alias) and mapping.